src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs

```diff
@@ -2114,6 +2115,99 @@ public void TestSqlCommandCancellationToken(string connection, int initalValue,
             }
         }
 
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
+        [ClassData(typeof(AEConnectionStringProvider))]
+        public void TestCustomKeyStoreProviderDuringAeQuery(string connectionString)
+        {
+            if (!SQLSetupStrategyAzureKeyVault.isAKVProviderRegistered)
+            {
+                SqlColumnEncryptionAzureKeyVaultProvider sqlColumnEncryptionAzureKeyVaultProvider =
+                    new SqlColumnEncryptionAzureKeyVaultProvider(new SqlClientCustomTokenCredential());
+
+                DummyKeyStoreProvider dummyProvider = new DummyKeyStoreProvider();
+
+                SqlConnection.RegisterColumnEncryptionKeyStoreProviders(customProviders:
+                    new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>(capacity: 2, comparer: StringComparer.OrdinalIgnoreCase)
+                    {
+                        { SqlColumnEncryptionAzureKeyVaultProvider.ProviderName, sqlColumnEncryptionAzureKeyVaultProvider},
+                        { DummyKeyStoreProvider.Name, dummyProvider}
+                    });
+
+                SQLSetupStrategyAzureKeyVault.isAKVProviderRegistered = true;
+            }
+
+            Dictionary<string, SqlColumnEncryptionKeyStoreProvider> requiredProvider =
+                new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>()
+                {
+                    { DummyKeyStoreProvider.Name, new DummyKeyStoreProvider() }
+                };
+
+            string notRequiredProviderName = "DummyProvider2";
+            Dictionary<string, SqlColumnEncryptionKeyStoreProvider> notRequiredProvider =
+                new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>()
+                {
+                    { notRequiredProviderName, new DummyKeyStoreProvider() }
+                };
+
+            ApiTestTable customKeyStoreProviderTable = _fixture.CustomKeyStoreProviderTestTable as ApiTestTable;
+            string cekHexWithHyphens = BitConverter.ToString(customKeyStoreProviderTable.columnEncryptionKey1.EncryptedValue);
+            int lengthOfTenBytesWithHyphens = 29;
```

Comment:
Where does this 29 come from?

Reply:
BitConverter.ToString returns a hex string delimited by hyphens, eg "AA-BB-CC...". 29 is the length of the last ten bytes and the hyphens. The same format is used in the exception message.

Reply:
OK. So each byte will be converted to a hexadecimal pair and separated by hyphens. Do you mind adding the format `AA-BB-CC...` in the comment?

Reply:
Figured out a better way using an overload of `BitConverter.ToString`.